### PR TITLE
Kitchen Food Buffs

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Nutrient
All nutrients grant a +25 stamina bonus when consumed. The duration of this buff is based on the amount consumed.

# Slop and Muck
Blending nutrients will turn the nutrients to slop. This only affects nutrient, not special chemicals like honey, flour, milk, and other chemicals. Slop fills you up, but causes a detriment to stamina (-25) and stamina regeneration for a variable amount of seconds. 

Blending meat will turn it into muck, which does the same thing as slop.

# Sugar
Consuming too much sugar can lead to sugar crashes, which causes a flat stamina loss (-20) for a variable duration based on the amount of sugar consumed. Sugar crashes only occur if the most sugar you've had in your system at one time exceeds 10 units.

Sugar is now a nutrient. It can damage antmen now, so be aware.

# Junkfood
Junkfood now contains various amounts of sugar, muck, and slop, save for SkrellSnacks which are expensive and healthy. Junkfood won't kill you anymore, but just gives you less stamina.

# Syringes
In order to prevent meme exploits, food can no longer be extracted by syringes. Uses the biogenerator vitamin pills if you really have to powergame for nutrients, smh.

### CODE REVIEW

**Comment by LordFowl** [comment]
This replaces a dumb system with a retarded system. There is no reason to use reagents like this.

**Comment by BurgerLUA** [comment]
I highly disagree and think that this is a huge, pointless overreaction that is somehow more whiny and more arbitrary than the combined crying of every sincere ooc cheapshot of junk food that I've witnessed while playing.

**Comment by BurgerLUA** [comment]
PR is reworked and fundamentally changed.

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -27,6 +27,9 @@
 	var/metabolism_min = 0.01 //How much for the medicine to be present in the system to actually have an effect.
 	var/list/conflicting_reagents //Reagents that conflict with this medicine, and cause adverse effects when in the blood.
 
+	var/list/container_whitelist = list()
```
Comment (Lohikar) [inline code/modules/reagents/Chemistry-Reagents.dm]: Do not make `= list()` vars on commonly instanced types, like reagents.

Comment (BurgerLUA) [reply code/modules/reagents/Chemistry-Reagents.dm]: I'll remove the feature entirely.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 2]
```diff
@@ -5171,3 +5152,13 @@
 /obj/item/weapon/reagent_containers/food/snacks/corn_dog/Initialize()
 	. = ..()
 	reagents.add_reagent("protein", 3)
+
+/obj/item/weapon/reagent_containers/food/snacks/meatloaf
```
[the remark this answers is not in the document]
Comment (Alberyk) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: If someone spawns one of those via other means, they will be empty, which should not really happen.

**Comment by Keinto** [comment]
Why not revert to how it used to be before this entire ordeal and JUST REMOVE MOST vending machines on the station?

I can't think of a simpler way to make the kitchen more appealing to the crew and at the same time not make the game dumber with debuffs and buffs from eating food in a 2 hour shift.

**Comment by BurgerLUA** [comment]
Vending machines need to exist to provide a quick, easy way to fill your hunger. It's convenience food, however this update makes it so that real food will be a better alternative than the current food without punishing people who eat junkfood too much.

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (BurgerLUA) [reply code/modules/reagents/reagent_containers/food/snacks.dm]: This was fixed.

**Comment by BurgerLUA** [comment]
Okay. This is now meant to work alongside the #4481 update.

Some additions:
- There is now synthetic protein. It doesn't make sense that junk food companies would skimp on the pastry but still provide quality meat.
- Slop and Muck are now positive, but not as good as nutrients. All junk food now contains synth nutrients or synth protein.

**Comment by skull132** [comment]
Since this is a conflict of interest for me, note that the following is the summary opinion of the other devs. /disclaimer

Anywho, they are either against it for the reason that it's still a bit excessive, or that it is now unnecessary considering #4481. So, at present time, this will be closed. If it turns out that #4481 is not on the spot, then this PR may be reconsidered at a later date.

**Pull request closed without merging**